assets/app/scripts/directives/logViewer.js

```diff
@@ -224,6 +228,35 @@ angular.module('openshiftConsole')
               goChromeless: logLinks.chromelessLink,
               restartLogs: streamLogs
             });
+
+            APIDiscovery
+              .getLoggingURL()
+              .then(function(url) {
+                // TODO: update lodash for _.get, would rather do this:
+                //var projectName = _.get($scope.context, 'project', 'metadata', 'name');
+                var projectName = $scope.context &&
+                                  $scope.context.project &&
+                                  $scope.context.project.metadata &&
+                                  $scope.context.project.metadata.name;
+
+                var hasComponents = projectName && $scope.name && url;
+                if(!hasComponents) {
+                  return;
+                }
+
+                angular.extend($scope, {
+                  canViewArchive: true,
+                  kibanaAuthUrl: $sce.trustAsResourceUrl(url+'/auth/token'),
+                  access_token: AuthService.UserStore().getToken(),
+                  // The archive URL violates angular's built in same origin policy.
+                  // Need to explicitly tell it to trust this location or it will throw errors.
+                  archiveLocation: $sce.trustAsResourceUrl(logLinks.archiveUri({
+                    namespace: projectName,
+                    podname: $scope.name,
+                    backlink: $window.location.href
```

Comment:
@liggitt @jwforres for our container logs we add metadata (such as container name, pod name, namespace, and k8s host) that we can filter/search on within Kibana.